### PR TITLE
DDF-5907 - Fix spellcheck Showing Results For

### PR DESCRIPTION
**Pull request by derekwilhelm**
#### What does this PR do?
Fixes the bug where enabling spellcheck causes a NullPointerException due to missing spellcheck response in the modified query.
#### Who is reviewing it? 
@cantstoptheunk
@bellcc
@pklinef

#### Select relevant component teams: 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@jlcsmith
@andrewkfiedler

#### How should this be tested?
1. After full build and install, enable spellcheck in the admin console
2. Build the spellcheck dictionary index in Solr
3. Upload some data
4. Create a search that enables spellcheck and search for an incorrectly spelled word
5. Assuming the misspelled word provides a spellcheck hit:
  a) In the logs, there should no longer be a NullPointerException 
  b) The "Did you mean" and "Showing results for" capability on the UI will be populated

#### What are the relevant tickets?
Fixes: #5907 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java [hunk 1]
```diff
@@ -260,7 +260,8 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
         addDocsToResults(docs, results);
 
         if (userSpellcheckIsOn && solrSpellcheckHasResults(solrResponse)) {
-          query.set("q", findQueryToResend(query, solrResponse));
+          Collation collation = findQueryToResend(query, solrResponse);
```
Comment (cantstoptheunk) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java]: ✏️ Possibly worth changing the method name to maybe something like findBestCollation or something that matches the return value more appropriately.

**Comment by derekwilhelm** [comment]
build now

**Comment by derekwilhelm** [comment]
build now

**Comment by jlcsmith** [comment]
build now

**Comment by jlcsmith** [comment]
build now

**Comment by cantstoptheunk** [comment]
Hero Successful:
Uploaded resources, and misspelled queries and got expected results. Screenshot below shows a query with two parameters (originale, and normale). The resource suggestion shows up, however this is dealt with a plug @derekwilhelm made in a downstream project to ignore that field.

However, may be worth considering bringing that plugin upstream.

Screen Shot 2020-03-27 at 9 52 34 AM

**Pull request merged**